src/js/simpleSmtpServer.js

```diff
@@ -0,0 +1,84 @@
+// A module to wire in simplesmtp to handle incoming messages.
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.test.mail.smtp.simpleSmtpServer");
+
+var simplesmtp = require("simplesmtp");
+var fs         = require("fs");
+
+gpii.test.mail.smtp.simpleSmtpServer.handleStartData = function (that, connection) {
+    var timestamp = (new Date()).getTime();
+    that.options.messageFile = that.options.config.outputDir + "/message-" + timestamp + ".txt";
+    connection.saveStream = fs.createWriteStream(that.options.messageFile);
+};
+
+gpii.test.mail.smtp.simpleSmtpServer.handleData = function (that, connection, chunk) {
+    connection.saveStream.write(chunk);
+};
+
+gpii.test.mail.smtp.simpleSmtpServer.handleDataReady = function (that, connection, callback) {
+    connection.saveStream.end();
+
+    that.events.messageReceived.fire(that, connection);
+
+    callback(null, that.options.config.queueId);
```

Comment:
Instead of being applied in the same callback, this should be registered in a separate listener with priority: "last" to ensure that any other listeners that users register to this event are notified before we call the mysterious "callback" supplied from the native dataReady event. At least, this is the purpose I understand for this callback - I assume it is intended as a notification to the implementation that we have finished all our processing for the data it supplied.